Doc/library/re.rst

```diff
@@ -1357,9 +1357,9 @@ Since :meth:`~Pattern.match` and :meth:`~Pattern.search` return ``None``
 when there is no match, you can test whether there was a match with a simple
 ``if`` statement::
 
-   match = re.search(pattern, string)
-   if match:
-       process(match)
+   m = re.search(pattern, string)
+   if m:
+       process(m)
```

Comment:
To promote good practice, we could suggest

```suggestion
   m = re.search(pattern, string)
   if m is not None:
       process(m)
```

If we want to suggest using assignment expressions, we could also use:

```suggestion
   if m := re.search(pattern, string):
       process(m)
```

Or both! (I probably wouldn't go for this...)

```suggestion
   if (m := re.search(pattern, string)) is not None:
       process(m)
```

A

Reply:
Having `is not None` does not seem to me better practice. There are two possibly relevant parts of PEP 8:

> Also, beware of writing `if x` when you really mean `if x is not None` – e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

and

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false:
> ```python
> # Correct:
> if not seq:
> if seq:
> ```
> ```python
> # Wrong:
> if len(seq):
> if not len(seq):
> ```

Now, the first does not apply here, because, as the documentation immediately before the example says,

> _Match objects always have a boolean value of `True`._ Since [match()](https://docs.python.org/3/library/re.html#re.Pattern.match) and [search()](https://docs.python.org/3/library/re.html#re.Pattern.search) return `None` when there is no match, you can test whether there was a match with a simple `if` statement [Emphasis added]

On the other hand, the spirit of the second excerpt from PEP 8 argues for availing of truthiness and falsiness and thus against the redundant `is not None`.